cmd/tpm-manager/tpm-manager.go

```diff
@@ -0,0 +1,507 @@
+	if newstate == true {
+		trustNode(newnode)
+	} else {
+		invalidateNode(newnode)
+	}
+	newnode.ObjectMeta.Annotations[LogState] = node.ObjectMeta.Annotations[LogState]
+	newnode, err = manager.client.Nodes().Update(newnode)
+	if err != nil {
+		glog.Errorf("Unable to update node state for %s: %v", node.Name, err)
+		return
+	}
+}
+
+func verifyAllNodes() {
+	nodes, err := manager.client.Nodes().List(api.ListOptions{})
+	if err != nil {
+		glog.Errorf("Unable to obtain list of nodes")
+		return
+	}
+	for _, node := range nodes.Items {
+		verifyAndUpdate(&node)
+	}
+}
+
+func reverify() {
+	for {
+		select {
+		case <-time.After(time.Duration(manager.recurring) * time.Second):
+			verifyAllNodes()
+		case <-manager.recurringChan:
+		}
+	}
+}
+
+type TPMManager struct {
+	Master        string
+	Kubeconfig    string
+	tpmhandler    tpm.TPMHandler
+	pcrConfig     string
+	pcrConfigDir  string
+	allowUnknown  bool
+	recurring     int
+	client        *client.Client
+	policyTimer   *time.Timer
+	leaderelect   componentconfig.LeaderElectionConfiguration
+	recurringChan chan int
+	pcrConfigs    []map[string]tpm.PCRConfig
+}
+
+var manager TPMManager
+
+func addFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&manager.Master, "master", manager.Master, "The address of the Kubernetes API server (overrides any value in kubeconfig)")
+	fs.StringVar(&manager.Kubeconfig, "kubeconfig", manager.Kubeconfig, "Path to kubeconfig file with authorization and master location information.")
+	fs.StringVar(&manager.pcrConfig, "pcrConfig", manager.pcrConfig, "Path to a single PCR config file")
+	fs.StringVar(&manager.pcrConfigDir, "pcrConfigDir", manager.pcrConfigDir, "Path to a PCR config directory")
+	fs.BoolVar(&manager.allowUnknown, "allowUnknown", false, "Allow unknown TPMs to join the cluster")
+	fs.IntVar(&manager.recurring, "reverify", 0, "Periodocally reverify nodes after this many seconds")
+}
+
+func updateConfig(configmap *api.ConfigMap) {
+	if configmap.Data["allowunknown"] != "" {
+		allowUnknown, err := strconv.ParseBool(configmap.Data["allowunknown"])
+		if err == nil {
+			manager.allowUnknown = allowUnknown
+		}
+	}
+	if configmap.Data["reverify"] != "" {
+		reverify, err := strconv.Atoi(configmap.Data["reverify"])
+		if err == nil {
+			manager.recurring = reverify
+			// Trigger the reverification logic. If it's already in the
+			// middle of reverifying, drop the event - it'll handle it
+			// at the end of reverification.
+			select {
+			case manager.recurringChan <- reverify:
+			default:
+			}
+		}
+	}
+}
+
+func run(stop <-chan struct{}) {
+	_, nodeController := framework.NewInformer(
+		&cache.ListWatch{
+			ListFunc: func(options api.ListOptions) (runtime.Object, error) {
+				return manager.client.Nodes().List(options)
+			},
+			WatchFunc: func(options api.ListOptions) (watch.Interface, error) {
+				return manager.client.Nodes().Watch(options)
+			},
+		},
+		&api.Node{},
+		controller.NoResyncPeriodFunc(),
```

Comment:
I believe it's possible to miss events (could be wrong, might want to check with @hongchaodeng).

So if an add event is missed, this will never resync the cache (or fire the updateFunc so you can re-check state). Not sure it matters in your use-case -- just wanted to make note.